ambry-commons/src/main/java/com.github.ambry.commons/BlobIdFactory.java

```diff
@@ -33,6 +33,6 @@ public BlobIdFactory(ClusterMap clusterMap) {
 
   @Override
   public StoreKey getStoreKey(DataInputStream value) throws IOException {
-    return new BlobId(value, clusterMap);
+    return BlobId.fromDataInputStream(value, clusterMap);
```

Comment:
There was some discussion that there should be a `static` method that does the des job, and building from scratch using constructor. I guess no such rule exists, so I will change it to constructor.